kratos/utilities/beam_math_utilities.hpp

```diff
@@ -539,46 +661,47 @@ template<class TDataType> class BeamMathUtils
   //*****************************************************************************
 
   /**
-   * Deffault expression for GID local beam axes:: E3 is considered the local beam axial direction
-   * @param rLocalZ: Local Beam axis director vector E3 (input parameter) (output parameter)
-   * @param rLocalX: Local Beam axis director vector E1 (output parameter)
+   * Deffault expression for GID local beam axes:: E1 is considered the local beam axial direction
+   * @param rLocalX: Local Beam axis director vector E1 (input parameter) (output parameter)
    * @param rLocalY: Local Beam axis director vector E2 (output parameter)
+   * @param rLocalZ: Local Beam axis director vector E3 (output parameter)
    */
-  static inline  void CalculateLocalAxesVectors(VectorType& rLocalZ, VectorType& rLocalX, VectorType& rLocalY)
+  template<class TVector3>
+  static inline  void CalculateLocalAxesVectors(TVector3& rLocalX, TVector3& rLocalY, TVector3& rLocalZ)
   {
 
     KRATOS_TRY
 
-    VectorType GlobalY = ZeroVector(3);
+    TVector3 GlobalY = ZeroVector(3);
     GlobalY[1]=1.0;
 
-    VectorType GlobalZ = ZeroVector(3);
+    TVector3 GlobalZ = ZeroVector(3);
     GlobalZ[2]=1.0;
 
-    // local z-axis (e3_local) is the beam axis
-    double VectorNorm = MathUtilsType::Norm(rLocalZ);
+    // local x-axis (e1_local) is the beam axis
+    double VectorNorm = MathUtilsType::Norm(rLocalX);
     if( VectorNorm != 0)
-      rLocalZ /= VectorNorm;
+      rLocalX /= VectorNorm;
     
-    // local x-axis (e1_local)  
+    // local y-axis (e2_local)  
     double tolerance = 1.0/64.0;
-    if(fabs(rLocalZ[0])< tolerance && fabs(rLocalZ[1])< tolerance){
-      rLocalX = MathUtilsType::CrossProduct(GlobalY, rLocalZ);
+    if(fabs(rLocalX[0])< tolerance && fabs(rLocalX[1])< tolerance){
+      rLocalY = MathUtilsType::CrossProduct(GlobalY, rLocalX);
     }
     else{
-      rLocalX = MathUtilsType::CrossProduct(GlobalZ, rLocalZ);
+      rLocalY = MathUtilsType::CrossProduct(GlobalZ, rLocalX);
     }
     
-    VectorNorm = MathUtilsType::Norm(rLocalX);
+    VectorNorm = MathUtilsType::Norm(rLocalY);
     if( VectorNorm != 0)
-      rLocalX /= VectorNorm;
+      rLocalY /= VectorNorm;
     
-    // local y-axis (e2_local)
-    rLocalY = MathUtilsType::CrossProduct(rLocalZ,rLocalX);
+    // local z-axis (e3_local)
+    rLocalZ = MathUtilsType::CrossProduct(rLocalX,rLocalY);
```

Comment:
same

          MathUtilsType::UnitCrossProduct(rLocalZ,rLocalX, rLocalY)

actually since this is correctly implemented it makes me think that in the two previous ones you assumed that
the CrossProduct was working "correctly", however most probably you are currently gettign the opposite sign

Reply:
I have to check it, I mostly used symmetrical sections.